src/test/java/org/mule/test/oauth/AbstractOAuthTestCase.java

```diff
@@ -78,7 +88,27 @@ public void setupServices() throws Exception {
     final InputStreamHttpEntity httpEntity = mock(InputStreamHttpEntity.class);
     when(httpEntity.getContent()).thenReturn(new ReaderInputStream(new StringReader("")));
     when(httpResponse.getEntity()).thenReturn(httpEntity);
-    when(httpClient.sendAsync(any(), any())).thenReturn(supplyAsync(() -> httpResponse));
+    when(httpClient.sendAsync(any(), any())).thenAnswer(invocation -> {
+
+      final CompletableFuture<HttpResponse> httpResponseFuture = new CompletableFuture<>();
+
+      httpClientCallbackExecutor.execute(() -> {
+        try {
+          sleep(10);
```

Comment:
naaaaaaaaa

Reply:
test code. It i were to do this with a latch, the countdown would need to be in production code :S